src/connector/src/sink/doris_starrocks_connector.rs

```diff
@@ -337,3 +378,234 @@ impl InserterInner {
+        ))
+    }
+
+    pub fn build_prepare_request_sender(&self, label: String) -> Result<MetaRequestSender> {
+        let request = self.build_request(self.url_prepare.clone(), Method::POST, label)?;
+        Ok(MetaRequestSender::new(
+            self.client.clone(),
+            request,
+            self.fe_host.clone(),
+        ))
+    }
+
+    pub fn build_commit_request_sender(&self, label: String) -> Result<MetaRequestSender> {
+        let request = self.build_request(self.url_commit.clone(), Method::POST, label)?;
+        Ok(MetaRequestSender::new(
+            self.client.clone(),
+            request,
+            self.fe_host.clone(),
+        ))
+    }
+
+    pub fn build_rollback_request_sender(&self, label: String) -> Result<MetaRequestSender> {
+        let request = self.build_request(self.url_rollback.clone(), Method::POST, label)?;
+        Ok(MetaRequestSender::new(
+            self.client.clone(),
+            request,
+            self.fe_host.clone(),
+        ))
+    }
+
+    pub async fn build_txn_inserter(&self, label: String) -> Result<InserterInner> {
+        let request = self.build_request(self.url_load.clone(), Method::PUT, label)?;
+        let mut request_for_redirection = request.try_clone().ok_or_else(|| {
+            SinkError::DorisStarrocksConnect(anyhow::anyhow!("Can't clone request"))
+        })?;
+
+        let resp = self
+            .client
+            .execute(request)
+            .await
+            .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
+
+        let be_url = try_get_be_url(&resp, self.fe_host.clone())?.ok_or_else(|| {
+            SinkError::DorisStarrocksConnect(anyhow::anyhow!("Can't get doris BE url",))
+        })?;
+        *request_for_redirection.url_mut() = be_url;
+
+        let (sender, receiver) = tokio::sync::mpsc::unbounded_channel();
+        let body = Body::wrap_stream(
+            tokio_stream::wrappers::UnboundedReceiverStream::new(receiver).map(Ok::<_, Infallible>),
+        );
+        *request_for_redirection.body_mut() = Some(body);
+
+        let client = self.client.clone();
+        let handle: JoinHandle<Result<Vec<u8>>> = tokio::spawn(async move {
+            let response = client
+                .execute(request_for_redirection)
+                .await
+                .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
+
+            let status = response.status();
+            let raw = response
+                .bytes()
+                .await
+                .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?
+                .into();
+
+            if status == StatusCode::OK {
+                Ok(raw)
+            } else {
+                Err(SinkError::DorisStarrocksConnect(anyhow::anyhow!(
+                    "Failed connection {:?},{:?}",
+                    status,
+                    String::from_utf8(raw)
+                        .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?
```

Comment:
Internal `err` was lost. Please either pass the err as `source` (recommended) or use `err.as_report()` to get the proper error message.

See also https://www.notion.so/risingwave-labs/A-Guide-to-Error-Handling-that-Just-Works-Part-I-e108f8d0845b4bddaa0e5843791a8cac

Reply:
The owner of the doc is vacationing. Let me export it. 
[A_Guide_to_Error_Handling_that_Just_Works_(Part_I).pdf](https://github.com/risingwavelabs/risingwave/files/15385569/24ed19fe-66a2-446d-8b8b-2cb9bf501eea_A_Guide_to_Error_Handling_that_Just_Works_.Part_I.pdf)

Reply:
I have replaced `SinkError::DorisStarrocksConnect(err.into())` with `SinkError::DorisStarrocksConnect(anyhow!(err))` to keep the uniformity with other sinks.